python/pyspark/sql/context.py

```diff
@@ -72,8 +72,13 @@ def __init__(self, sparkContext, sparkSession=None, jsqlContext=None):
         self._sc = sparkContext
         self._jsc = self._sc._jsc
         self._jvm = self._sc._jvm
+
         if sparkSession is None:
-            sparkSession = SparkSession(sparkContext)
+            if sparkContext is SparkContext._active_spark_context:
+                sparkSession = SparkSession.builder.getOrCreate()
```

Comment:
I don't know enough to usefully review, but is the better fix to call the getOrCreate that lets you pass a context and use that for both cases? I don't know if it is the same behavior but seems more natural to use the logic in SparkSession for this.